PowerShellEditorServices.build.ps1

```diff
@@ -254,45 +256,62 @@ Task TestE2EPowerShell -If (-not $script:IsNix) Build, SetupHelpForTests, {
+        Write-Build DarkRed 'Skipping Constrained Language Mode tests as they must be ran in an elevated process'
         return
     }
 
     try {
-        Write-Build DarkGreen "Running end-to-end tests in Constrained Language Mode"
-        [System.Environment]::SetEnvironmentVariable("__PSLockdownPolicy", "0x80000007", [System.EnvironmentVariableTarget]::Machine)
+        Write-Build DarkGreen 'Running end-to-end tests in Constrained Language Mode'
+        [System.Environment]::SetEnvironmentVariable('__PSLockdownPolicy', '0x80000007', [System.EnvironmentVariableTarget]::Machine)
         Invoke-BuildExec { & dotnet $script:dotnetTestArgs $script:NetFramework.PS74 }
     } finally {
-        [System.Environment]::SetEnvironmentVariable("__PSLockdownPolicy", $null, [System.EnvironmentVariableTarget]::Machine)
+        [System.Environment]::SetEnvironmentVariable('__PSLockdownPolicy', $null, [System.EnvironmentVariableTarget]::Machine)
     }
 }
 
 Task TestE2EPowerShellCLM -If (-not $script:IsNix) Build, SetupHelpForTests, {
     Set-Location ./test/PowerShellEditorServices.Test.E2E/
-    $env:PWSH_EXE_NAME = "powershell"
+    $env:PWSH_EXE_NAME = 'powershell'
 
-    if (-not [Security.Principal.WindowsIdentity]::GetCurrent().Owner.IsWellKnown("BuiltInAdministratorsSid")) {
-        Write-Build DarkRed "Skipping Constrained Language Mode tests as they must be ran in an elevated process"
+    if (-not [Security.Principal.WindowsIdentity]::GetCurrent().Owner.IsWellKnown('BuiltInAdministratorsSid')) {
+        Write-Build DarkRed 'Skipping Constrained Language Mode tests as they must be ran in an elevated process'
         return
     }
 
     try {
-        Write-Build DarkGreen "Running end-to-end tests in Constrained Language Mode"
-        [System.Environment]::SetEnvironmentVariable("__PSLockdownPolicy", "0x80000007", [System.EnvironmentVariableTarget]::Machine)
+        Write-Build DarkGreen 'Running end-to-end tests in Constrained Language Mode'
+        [System.Environment]::SetEnvironmentVariable('__PSLockdownPolicy', '0x80000007', [System.EnvironmentVariableTarget]::Machine)
         # TODO: See https://github.com/PowerShell/vscode-powershell/issues/3886
         # Inheriting the module path for powershell.exe breaks things!
         $originalModulePath = $env:PSModulePath
-        $env:PSModulePath = ""
+        $env:PSModulePath = ''
         Invoke-BuildExec { & dotnet $script:dotnetTestArgs $script:NetFramework.PS74 }
     } finally {
-        [System.Environment]::SetEnvironmentVariable("__PSLockdownPolicy", $null, [System.EnvironmentVariableTarget]::Machine)
+        [System.Environment]::SetEnvironmentVariable('__PSLockdownPolicy', $null, [System.EnvironmentVariableTarget]::Machine)
         $env:PSModulePath = $originalModulePath
     }
 }
 
+Task BuildIfChanged.Init -Before BuildIfChanged {
+    [bool]$SCRIPT:ChangesDetected = $false
+}
+
+Task BuildIfChanged -Inputs {
```

Comment:
Nice, I'd briefly looked into this before wanting to leverage Invoke-Build's incremental build support but couldn't quite figure it out. This is clean!

Reply:
I think there's a minor bug here where occasionally dotnet doesn't actually publish the new DLL if the end result content hasn't been byte altered, resulting in it building every time because the source is newer than the destination, but a clean fixes that pretty quick so I haven't gone too deep into it.